test/integration/consul-container/libs/service/gateway.go

```diff
@@ -101,6 +102,10 @@ func (g gatewayContainer) GetAdminAddr() (string, int) {
 	return "localhost", g.adminPort
 }
 
+func (g gatewayContainer) GetPort(port int) int {
+	return g.portMappings[port]
+}
```

Comment:
Would it make sense to modify this function to return the boolean representing if the port key exists?

```suggestion
func (g gatewayContainer) GetPort(port int) (int, bool) {
	return g.portMappings[port]
}
```

Or if a port missing is an error, 

```suggestion
func (g gatewayContainer) GetPort(port int) (int, error) {
	p, ok := g.portMappings[port]
	if !ok {
	    return 0, fmt.Errorf("port does not exist")
	}
	return p, nil
}
```

Reply:
Possibly, but considering that this would involve modifying other segments of the preexisting tests, I'd rather not on this PR